### PR TITLE
ROC-6121 Create citizen claim endpoint for ioc

### PR DESCRIPTION
**Pull request by simad**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-6121

### Change description ###

Adding endpoint to create a claim, after initiating / resuming a payment

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/ClaimController.java [hunk 1]
```diff
@@ -133,6 +133,17 @@ public CreatePaymentResponse resumePayment(
         return claimService.resumePayment(authorisation, claimData);
     }
 
+    @PutMapping(value = "/{submitterId}/create-citizen-claim", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ApiOperation("Creates a citizen claim")
+    public Claim createClaim(
+        @Valid @NotNull @RequestBody ClaimData claimData,
+        @RequestHeader(HttpHeaders.AUTHORIZATION) String authorisation
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/ClaimController.java]: `front-end` passes list of `features` which are enabled on the claim, and we save as part of that claim, we have those missing here.

Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/ClaimController.java]: 👍

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 2]
```diff
@@ -219,6 +218,27 @@ public CreatePaymentResponse resumePayment(
             .build();
     }
 
+    @LogExecutionTime
+    @Transactional(transactionManager = "transactionManager")
+    public Claim saveCitizenClaim(
+        String authorisation,
+        ClaimData claimData,
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: We are getting this ClaimData from our endpoint, but we are not storing this in CCD.

Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: yeah, I was imagining the claim data to be the same as the one we already have stored in ccd after the initiate/resume payment. But you're right, let's save it too

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: Thanks

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 3]
```diff
@@ -437,9 +424,26 @@ private Claim buildClaimFrom(
...
+        if (asyncEventOperationEnabled) {
+            eventProducer.createClaimCreatedEvent(
+                savedClaim,
+                user.getUserDetails().getFullName(),
+                authorisation
+            );
+        } else {
+            eventProducer.createClaimIssuedEvent(
+                savedClaim,
+                "to-be-removed",
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: This should have been removed ;)

**Pull request merged**